 <launch>
    <node name="track_apriltag_landing_velocity" pkg="dji_m100_landing" type="track_apriltag_landing_velocity" output="screen" >
    <param name="apriltags_36h11_sub" value="/tag_detections" />
    </node>

    <node pkg="dji_m100_landing" type="position_track_velocity_controller" name="position_track_velocity_controller_node" output="screen">
      </node>

   <node pkg="robot_pose_ekf" type="robot_pose_ekf" name="robot_pose_ekf">
    <param name="output_frame" value="odom"/>
    <param name="freq" value="30.0"/>
    <param name="sensor_timeout" value="1.0"/>
    <param name="odom_used" value="true"/>
    <param name="imu_used" value="true"/>
    <param name="vo_used" value="true"/>
    <param name="debug" value="false"/>
    <param name="self_diagnose" value="false"/>

    <remap from="odom" to="dji_sdk/odometry" />
    <remap from="imu" to="dji_sdk/imu" />
    <remap from="vo" to="dji_sdk/imu" />
  </node>

 <!--
  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
    <param name="video_device" value="/dev/video1" />
    <param name="image_width" value="640" />
    <param name="image_height" value="480" />
    <param name="pixel_format" value="yuyv" />
    <param name="camera_frame_id" value="usb_cam" />
    <param name="io_method" value="mmap"/>
  </node>
-->

    <!--
    <include file="$(find apriltag_ros)/launch/continuous_detection.launch">
 		<arg name="image_rect_topic" default="/usb_cam/image_rect" />
		<arg name="camera_info_topic" default="/usb_cam/camera_info" />
		<arg name="camera_frame" default="usb_cam" />
    </include>


   <include file="$(find apriltag_ros)/launch/continuous_detection.launch">
 		<arg name="image_rect_topic" default="/guidance_down/left/image_rect" />
		<arg name="camera_info_topic" default="/guidance_down/left/camera_info" />
		<arg name="camera_frame" default="guidance_down" />
    </include>-->

    <node pkg="tf" type="static_transform_publisher" name="usb_to_base" args="0 0 -0.1 -0.7071068 0.7071068 0 0 /usb_cam /base_link 100"/>

     <node pkg="landing_starter" type="landing_starter" name="landing_starter" output="screen"/>


    <node ns="dji_m100_landing" name="position_track_pid_velocity_x_controller" pkg="pid" type="controller">
      <param name="node_name" value="apriltag_tracking_x_controller" />
      <param name="Kp" value="0.9" />
      <param name="Ki" value="0.015 " />
      <param name="Kd" value="0.1" />
      <param name="converge_threshold" value="0.05" />
      <param name="upper_limit" value="30" />
      <param name="lower_limit" value="-30" />
      <param name="windup_limit" value="100" />
      <param name="diagnostic_period" value="0.25" />
      <param name="max_loop_frequency" value="105.0" />
      <param name="min_loop_frequency" value="95.0" />
      <remap from="setpoint" to="/teamc/position_track_x/setpoint" />
      <param name="topic_from_plant" value="/teamc/position_track_x/state" />
      <param name="topic_from_controller" value="/teamc/position_track_x/control_effort" />
     </node>

    <node ns="dji_m100_landing" name="position_track_pid_velocity_y_controller" pkg="pid" type="controller"  >
      <param name="node_name" value="apriltag_tracking_y_controller" />
      <param name="Kp" value="0.9" />
      <param name="Ki" value="0.015" />
      <param name="Kd" value="0.1" />
      <param name="converge_threshold" value="0.05" />
      <param name="upper_limit" value="3" />
      <param name="lower_limit" value="-30" />
      <param name="windup_limit" value="100" />
      <param name="max_loop_frequency" value="105.0" />
      <param name="min_loop_frequency" value="95.0" />
      <remap from="setpoint" to="/teamc/position_track_y/setpoint" />
      <param name="topic_from_plant" value="/teamc/position_track_y/state" />
      <param name="topic_from_controller" value="/teamc/position_track_y/control_effort" />
    </node>




    <node ns="dji_m100_landing" name="position_track_pid_yaw_controller" pkg="pid" type="controller" >
      <param name="node_name" value="apriltag_tracking_yaw_controller" />
      <param name="Kp" value="0.9" />
      <param name="Ki" value="0.015" />
      <param name="Kd" value="0.1" />
      <param name="converge_threshold" value="1" />
      <param name="upper_limit" value="0.15" />
      <param name="lower_limit" value="-0.15" />
      <param name="windup_limit" value="1100" />
      <param name="cutoff_frequency" value="20" />
      <param name="diagnostic_period" value="0.25" />
      <param name="max_loop_frequency" value="105.0" />
      <param name="min_loop_frequency" value="95.0" />
      <remap from="setpoint" to="/teamc/position_track_yaw/setpoint" />
      <param name="topic_from_plant" value="/teamc/position_track_yaw/state" />
      <param name="topic_from_controller" value="/teamc/position_track_yaw/control_effort" />
     </node>
 </launch>
